inc/admin.php

```diff
@@ -48,18 +48,19 @@ function delete_pending_comments_icon() {
  */
 function nkdeletepending_options_page() {
 	$magic_string = __("I am sure I want to delete all pending comments and realize this can't be undone", 'delete-pending-comments' );
-	if ( current_user_can( 'manage_options' ) ) { ?>
+	if ( current_user_can( 'manage_options' ) ) {
+		// Limit. Timeout with 10k comments...
+		// We only need this to see if there are any pending anyway
+		$pending_comment_ids    = get_comments( 'status=hold&fields=ids' );
```

Comment:
Use direct SQL query to get the comment count. Don't use `get_comments`.

Reply:
Fixed.